src/ImageStack.cpp

```diff
@@ -50,75 +51,104 @@ int ImageStack::addImage(Image && i) {
             for (size_t i = 0; i < 6; ++i) {
-                fcrow[i] = params.FC(i, y);
+                fc_row[i] = params.FC(i, y);
             }
             size_t x = 0;
             for (; x < width - 5; x+=6) {
                 for(size_t j = 0; j < 6; ++j) {
-                    uint16_t v = brightest(x + j, y);
-                    ++histogramsThr[fcrow[j]][v];
+                    uint16_t luminance_val = brightest_image(x + j, y);
+                    ++histograms_thr[fc_row[j]][luminance_val];
                 }
             }
             // remaining pixels
             for (size_t j = 0; x < width; ++x, ++j) {
-                uint16_t v = brightest(x, y);
-                ++histogramsThr[fcrow[j]][v];
+                uint16_t luminance_val = brightest_image(x, y);
+                ++histograms_thr[fc_row[j]][luminance_val];
             }
         }
         #pragma omp critical
         {
-            for (int c = 0; c < 4; ++c) {
-                for (std::vector<size_t>::size_type i = 0; i < histograms[c].size(); ++i) {
-                    histograms[c][i] += histogramsThr[c][i];
+            for (int c_channel = 0; c_channel < 4; ++c_channel) {
+                std::vector<size_t>::size_type histogram_size, luminance_val;
+                histogram_size = histograms[c_channel].size();
+                for (luminance_val = 0; luminance_val < histogram_size; ++luminance_val) {
+                    histograms[c_channel][luminance_val] += histograms_thr[c_channel][luminance_val];
                 }
             }
         }
     }
 
-    const size_t threshold = width * height / 10000;
+    // used to ignore luminances with frequentcy less htan 0.0001
+    const size_t occurance_threshold = width * height / 10000;
 
-    uint16_t maxPerColor[4] = {0, 0, 0, 0};
+    // maximum "significant" luminance per color channel
+    uint16_t max_lum_per_color[4] = {0, 0, 0, 0};
 
-    for (int c = 0; c < 4; ++c) {
-        for (int i = histograms[c].size() - 1; i >= 0; --i) {
-            const size_t v = histograms[c][i];
-            if (v > threshold) {
-                maxPerColor[c] = i;
+    for (int c_channel = 0; c_channel < 4; ++c_channel) {
+        // start from the highest value in the histograms (brightest)
+        for (int luminance_val = histograms[c_channel].size() - 1; luminance_val >= 0; --luminance_val) {
+            const size_t frequency = histograms[c_channel][luminance_val];
+            // ignore if it has a low occurance (frequency) in the image
+            if (frequency > occurance_threshold) {
+                max_lum_per_color[c_channel] = luminance_val;
                 break;
             }
         }
     }
 
 
-    uint16_t maxPerColors = std::max(maxPerColor[0], std::max(maxPerColor[1],std::max(maxPerColor[2], maxPerColor[3])));
-    satThreshold = params.max == 0 ? maxPerColors : params.max;
+    uint16_t max_luminance = std::max(max_lum_per_color[0], std::max(max_lum_per_color[1],std::max(
```

Comment:
C++11 has a [`std::max(std::initializer_list<T> ilist)`](https://en.cppreference.com/w/cpp/algorithm/max) which helps here. `std::uint16_t` and `const`?